### ISSUE TITLE
Optical kerning return 0.

### ISSUE THREAD

**Issue by quangxuagoluan** [Agamnentzar/ag-psd#198]
Optical kerning is set for spacing, but the returned value is 0. Is it possible to return a different value to distinguish between optical kerning being set and the value being set to 0?
![image](https://github.com/user-attachments/assets/1700d99e-2f88-4bbb-b4f3-4cf98650faa6)
[text.zip](https://github.com/user-attachments/files/16703643/text.zip)

**Comment by Agamnentzar** [comment]
Unfortunately it seems that this information is saved in `psd.engineData` field, there's no documentation of that data and it's structure is very cryptic. If you just need to read that field and don't need to update it you can try my unfinished function for decoding it and try to figure out how to use it:

```
const data = toByteArray(psd.engineData);
const engineData = parseEngineData(data);
const engineData2 = decodeEngineData2(engineData);
```

Inside `engineData2` structure you'll find `Kerning?` field with value `2` I think that's how it's flagging "Optical kerning"

![image](https://github.com/user-attachments/assets/3d7f0814-6595-4493-a2eb-bdfe8c280a59)

**Comment by quangxuagoluan** [comment]
> Unfortunately it seems that this information is saved in `psd.engineData` field, there's no documentation of that data and it's structure is very cryptic. If you just need to read that field and don't need to update it you can try my unfinished function for decoding it and try to figure out how to use it:
> 
> ```
> const data = toByteArray(psd.engineData);
> const engineData = parseEngineData(data);
> const engineData2 = decodeEngineData2(engineData);
> ```
> 
> Inside `engineData2` structure you'll find `Kerning?` field with value `2` I think that's how it's flagging "Optical kerning"
> 
> ![image](https://private-user-images.githubusercontent.com/572298/360281268-3d7f0814-6595-4493-a2eb-bdfe8c280a59.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzOTU3MTcsIm5iZiI6MTcyNDM5NTQxNywicGF0aCI6Ii81NzIyOTgvMzYwMjgxMjY4LTNkN2YwODE0LTY1OTUtNDQ5My1hMmViLWJkZmU4YzI4MGE1OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyM1QwNjQzMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMWU5OTZiOGY1YjU5NWY4MTg1MTRlYjFlMWZmMzU0MTAwZjU0Yzg4Njg0MDZhMGE0NDIwYTM3MWRkNDExZjE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.p_z5Ci6YKj9dHz8gWqE2FvS86FvFgMQ5x1m8DljGAN0)

Referring to ResourceDict.StyleSheetSet[0].StyleSheetData.Kerning?, when kerning is set to metric, it returns 1; when set to optical, it returns 2; and when set to 0, it returns 0. Additionally, even if I individually set a different kerning value internally, the overall external value does not change. There may be multiple StyleSheetSet objects in ResourceDict, but the Kerning? value is consistent across all of them.

**Comment by quangxuagoluan** [comment]
<code>Kerning?</code> It appears in multiple places, and the values are different in each case. I've done a lot of testing and finally found a place where there's a difference. I created a text layer in a PSD file and then applied different kerning to it.
```js
const engineData = parseEngineData(data);
const engineData2 = decodeEngineData2(engineData);
const kerning = get(engineData2, 'EngineDict["Editors?"][0].Editor.StyleRun.RunArray[0].StyleSheet.StyleSheetData["Kerning?"]')
console.log(kerning, 'kerning');
// 0 - 0
// 1 - metric
// 2- optical
```
When kerning is set to metric, it returns 1; when set to optical, it returns 2; and when set to 0, it returns 0。

**Comment by Agamnentzar** [comment]
Each element in `Editors?` array seem to reference different text layer, but I'm not sure which one matches which layer, I couldn't find layer ID field anywhere in there that would link both, maybe they're just in the same order as the text layers.

**Comment by quangxuagoluan** [comment]
Yes, I found that this index is consistent with its own position.
<code>EngineDict["Editors?"]</code>
![image](https://github.com/user-attachments/assets/4f3659f7-f732-4bf7-b7ee-c0b48ab07e74)
<code>text.index</code>
![image](https://github.com/user-attachments/assets/1dd63831-e4e7-4571-822c-e9519d578997)
